tests/jballerina-unit-test/src/test/resources/test-src/closures/closure.bal

```diff
@@ -751,3 +752,47 @@ function funcAcceptingIntArgs(int... scores) returns int {
     };
     return fn();
 }
+
+string moduleLevel = "moduleText1 ";
+
+function errorConstructorReference1(string param1) returns error? {
+    string temp = "text " + param1;
+    var process = function(string param2) returns error? {
+        moduleLevel = moduleLevel + param2;
+        return error("An Error", tempString = temp, moduleString = moduleLevel);
+    };
+    return process("tempParam2");
+}
+
+function errorConstructorReference2() returns function() returns error? {
+    string temp = "text";
+    var process = function() returns error? {
+        return error("An Error", tempString = temp);
```

Comment:
Shall we add a param of a mutable value (like `string[]`) here, update it and then assert the error.

e.g.,
where `errorConstructorReference2` accepts a `string[] param` parameter, and this line is
```suggestion
        return error("An Error", tempString = temp, params = param);
```

```ballerina
string[] arr = ["hello"];
function() returns error res = errorConstructorReference2(arr);
arr.push("world");
// Call res here, validate that the detail mapping has ["hello", "world"] as the entry for `params`.
```

I've removed nil from the return type of the returned function.